opm/autodiff/MultisegmentWell_impl.hpp

```diff
@@ -738,6 +840,8 @@ namespace Opm
             average_density /= sum_kr;
 
             cell_perforation_pressure_diffs_[perf] = gravity_ * average_density * cell_perforation_depth_diffs_[perf];
+
+            Base::perf_densities_[perf] = average_density;
```

Comment:
I do not think MSW should use `perf_densities_`. It should stay within `StandardWell`.  MSW has different way in handling the fluid densities within wellbore.

Reply:
I agree that they are computed differently, but both methods computes as average perforation density.